pkg/apis/eksctl.io/v1alpha4/validation.go

```diff
@@ -135,5 +136,33 @@ func ValidateNodeGroup(i int, ng *NodeGroup) error {
 		return err
 	}
 
+	if err := validateNodeGroupSSH(ng.SSH); err != nil {
+		return errors.New("only one ssh public key can be specified per node-group")
+	}
+	return nil
+}
+
+func validateNodeGroupSSH(SSH *NodeGroupSSH) error {
+	if SSH == nil {
+		return nil
+	}
+	numSSHFlagsEnabled := countEnabledFlags(
+		SSH.PublicKeyPath,
+		SSH.PublicKey,
+		SSH.PublicKeyName)
+
+	if numSSHFlagsEnabled > 1 {
+		return errors.New("only one ssh public key can be specified per node-group")
+	}
 	return nil
 }
+
+func countEnabledFlags(flags ...*string) int {
```

Comment:
Let's try to use word flags when it's about the CLI flags, you can use 'fields' or 'params', whichever you like :)